backend/tests/services/test_local_device_command_service.py

```
@@ -324,6 +324,198 @@ def test_local_device_command_registry_default_includes_diagnostic_commands():
+    tmp_path, monkeypatch, capsys
+):
+    """workspace_tree should emit stable JSON metadata for direct children."""
+    import json
+
+    from app.services.device.command_registry import WORKSPACE_TREE_SCRIPT
+
+    (tmp_path / "backend").mkdir()
+    (tmp_path / "README.md").write_text("hello", encoding="utf-8")
+    monkeypatch.chdir(tmp_path)
+
+    exec(WORKSPACE_TREE_SCRIPT, {"__name__": "__main__"})
```

Ruff (0.15.15)

[error] Use of `exec` detected

(S102)